build-system/tasks/runtime-test/helpers.js

```diff
@@ -249,9 +289,402 @@ function isLargeRefactor() {
+      browsers: [
+        'SL_Chrome',
+        'SL_Firefox',
+        'SL_Edge_17',
+        'SL_Safari_12',
+        'SL_IE_11',
+        // TODO(amp-infra): Evaluate and add more platforms here.
+        //'SL_Chrome_Android_7',
+        //'SL_iOS_11',
+        //'SL_iOS_12',
+        'SL_Chrome_Beta',
+        'SL_Firefox_Beta',
+      ],
+    })
+    .set('saucelabs_lite', {browsers: ['SL_Safari_12', 'SL_Firefox']});
+
+  for (const key of Array.from(options.keys())) {
+    if (argv[key]) {
+      return options.get(key);
+    }
+  }
+}
+
+/**
+ * Prints help messages for args if tests are being run for local development.
+ */
+function maybePrintArgvMessages() {
+  if (argv.nohelp) {
+    return;
+  }
+
+  const argvMessages = {
+    safari: 'Running tests on Safari.',
+    firefox: 'Running tests on Firefox.',
+    ie: 'Running tests on IE.',
+    edge: 'Running tests on Edge.',
+    'chrome_canary': 'Running tests on Chrome Canary.',
+    saucelabs: 'Running integration tests on Sauce Labs browsers.',
+    saucelabs_lite: 'Running tests on a subset of Sauce Labs browsers.', // eslint-disable-line google-camelcase/google-camelcase
+    nobuild: 'Skipping build.',
+    watch:
+      'Enabling watch mode. Editing and saving a file will cause the' +
+      ' tests for that file to be re-run in the same browser instance.',
+    verbose: 'Enabling verbose mode. Expect lots of output!',
+    testnames: 'Listing the names of all tests being run.',
+    files: 'Running tests in the file(s): ' + cyan(argv.files),
+    integration:
+      'Running only the integration tests. Prerequisite: ' +
+      cyan('gulp dist --fortesting'),
+    unit: 'Running only the unit tests. Prerequisite: ' + cyan('gulp css'),
+    a4a: 'Running only A4A tests.',
+    compiled: 'Running tests against minified code.',
+    grep:
+      'Only running tests that match the pattern "' + cyan(argv.grep) + '".',
+    coverage: 'Running tests in code coverage mode.',
+    headless: 'Running tests in a headless Chrome window.',
+    // eslint-disable-next-line google-camelcase/google-camelcase
+    local_changes:
+      'Running unit tests directly affected by the files' +
+      ' changed in the local branch.',
+  };
+  if (argv.chrome_flags) {
+    log(
+      green('Launching'),
+      cyan(CHROMEBASE),
+      green('with flags'),
+      cyan(chromeFlags)
+    );
+  }
+  if (!isTravisBuild()) {
+    log(
+      green('Run'),
+      cyan('gulp help'),
+      green('to see a list of all test flags.')
+    );
+    log(green('⤷ Use'), cyan('--nohelp'), green('to silence these messages.'));
+    if (
+      !argv.unit &&
+      !argv.integration &&
+      !argv.files &&
+      !argv.a4a &&
+      !argv['local-changes']
+    ) {
+      log(green('Running all tests.'));
+      log(
+        green('⤷ Use'),
+        cyan('--unit'),
+        green('or'),
+        cyan('--integration'),
+        green('to run just the unit tests or integration tests.')
+      );
```

Comment:
I think this message (line 403-417) can be removed. Once upon a time, it used to be possible to run unit and integration tests together, then we started discouraging it, and now, you're cleanly separating them.

Reply:
Nice, love deleting code